src/Utils/request/api-request.ts

```diff
@@ -0,0 +1,38 @@
+import { RequestResult, Route } from "@/Utils/request/types";
+
+import request from "./request";
+
+interface QueryOptions<TData> {
+  pathParams?: Record<string, string>;
+  queryParams?: Record<string, string>;
+  silent?: boolean;
+  onResponse?: (result: RequestResult<TData>) => void;
+}
+
+/**
+ * Creates a TanStack Query compatible request function
+ */
+function createQuery<TData, TBody>(
+  route: Route<TData, TBody>,
+  options?: QueryOptions<TData>,
+) {
+  return async ({
+    signal,
+  }: {
+    signal: AbortSignal;
+  }): Promise<RequestResult<TData>> => {
+    return request(route, {
+      pathParams: options?.pathParams,
+      query: options?.queryParams,
+      signal,
+      silent: options?.silent,
+      onResponse: options?.onResponse,
+    });
+  };
```

Comment:
Can we create a new request fn. instead? That's almost similar to the existing request, but does only the request part and not all the notification and stuff?

And we can set the onError in the global query client instead:

Refer: https://tkdodo.eu/blog/react-query-error-handling#the-global-callbacks

Also we don't need `onResponse` in newer usages. Since useQuery itself provides `onSuccess` and `onError` callbacks as options.

Reply:
@rithviknishad I've pushed some changes for simplifying the new request function, using `onError` to handle errors globally. Please review the changes